meta-oe/recipes-core/sdbus-c++/sdbus-c++_0.7.2.bb

```diff
@@ -0,0 +1,46 @@
+SUMMARY = "sdbus-c++"
+DESCRIPTION = "High-level C++ D-Bus library designed to provide easy-to-use yet powerful API in modern C++"
+
+SECTION = "libs"
+
+LICENSE = "LGPLv2.1"
+LIC_FILES_CHKSUM = "file://${WORKDIR}/sdbus-cpp-${PV}/COPYING;md5=1803fa9c2c3ce8cb06b4861d75310742"
+
+inherit cmake pkgconfig systemd ptest python3native
+
+DEPENDS += "expat ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'systemd', 'meson-native ninja-native git-native gperf-native gettext-native util-linux glibc libcap', d)}"
```

Comment:
I think hardcoding dependency on 'glibc' is not right. Did you mean something else here ?
and I dont understand why these dependencies are conditional on systemd. They should be added
in general.

Reply:
Things are conditional because of flexibility. sdbus-c++and has two modes of building. One is when entire systemd ecosystem is available in the system. Then `libsystemd` shared library, the library with lower level D-Bus C implementation which sdbus-c++ depends on, is available, and will be used. But sdbus-c++ provides flexibility to be used on non-systemd systems (where there is no `libsystemd` library). In that case, sdbus-c++ build builds just the small needed part of entire systemd and incorporates it as part of sdbus-c++ lib, making sdbus-c++ independent of systemd. Hence this conditional logic in the recipe which checks whether systemd feature is on or off. The user of yocto doesn't care, they build sdbus-c++ into their OS and it works perfectly whether or not systemd ecosystem is present.

Reply:
Regarding glibc, see previous comment.

Reply:
When sdbus-c++ needs to build that sd-bus part of systemd, which is in libsystemd, it needs all those additional deps, like `meson-native ninja-native libpcap etc.`

Reply:
Okay and to be clear these dependencies are not needed when building without systemd

Reply:
> Okay and to be clear these dependencies are not needed when building without systemd

Yes, when building sdbus-c++ in a system where systemd feature is turned on (systemd is present), these dependencies are not selected, instead simply `systemd` dependency is selected.